cbits/odbc.c

```diff
@@ -102,6 +107,7 @@ EnvAndDbc *odbc_AllocEnvAndDbc(){
         envAndDbc->env = env;
         envAndDbc->dbc = dbc;
         envAndDbc->error = malloc(MAXBUFLEN);
+        envAndDbc->sqlState = malloc(MAXBUFLEN);
```

Comment:
I suppose you took this from the previous version of the code but SQLSTATE is a 5 character string so there's no need to allocate 1024 bytes when we need only 6. See e.g. https://docs.microsoft.com/en-us/sql/odbc/reference/develop-app/using-sqlgetdiagrec-and-sqlgetdiagfield?view=sql-server-ver15
Also from that example it looks like currently we have also a couple of other issues with error processing:
* there could be multiple diagnostic records
* [SQL_MAX_MESSAGE_LENGTH](https://github.com/microsoft/ODBC-Specification/blob/master/Windows/inc/sql.h#L74) is 512 which is also less than currently used MAXBUFLEN=1024

But those are out of the scope for this PR probably.
And yet if your goal is to improve error handling for SQL Server then those should be addressed too.

Reply:
> I suppose you took this from the previous version of the code but SQLSTATE is a 5 character...

Yes, I blindly followed the previous code. Thank you for pointing this out. I'll push a commit to fix this.

> there could be multiple diagnostic records

I'm not sure yet what these records actually mean. I need to spend some time reading and learning about them. Until then I cannot address this. 

> SQL_MAX_MESSAGE_LENGTH is 512...

I'll address this request in my next commit.

Thank you @qrilka